### ISSUE TITLE
YubiKey setup resulting in `Invalid security key` error

### ISSUE THREAD

**Issue by foss-** [mastodon/mastodon#23502]
### Steps to reproduce the problem

1. setup TOTP
2. in `Account > Two Factor Auth` click Add Security Key
3. enter Nickname an click `Add new security key`
4. enter device PIN and proceed with click on `Sign In`
5. YubiKey is prompted, tap it to confirm prompt

### Expected behaviour

Successful YubiKey setup as security key

### Actual behaviour

Error message `Invalid security key`

### Specifications

YubiKey NFC 5 USB-C
Firefox 109.1 with CTAP 2 enabled in `about:config`

**Comment by ClearlyClaire** [comment]
This works when CTAP2 is disabled, but I can reproduce the issue when enabling it. I'm unfortunately not very knowledgeable about WebAuthn, so I'm not sure exactly what is wrong.

The client-side part sends empty values for `credential.id` (and `credential.rawid`), which the server requires.

**Comment by ClearlyClaire** [comment]
I have tried updating `@github/webauthn-json` to latest version just in case, but it has not solved anything. For the record, I'm using a YubiKey 5 NFC (USB-A) on Firefox 109.0.

Every WebAuthn demo I could find fails either at registration or log-in when enabling CTAP2 so I suspect this is a Firefox issue. Of course, anyone with a better understanding of WebAuthn and CTAP2 is welcome to chime in and correct me if I'm wrong.

**Comment by neatchee** [comment]
I do not believe this issue is limited to Firefox as I'm also experiencing the same behavior on Chrome for Android.

USB-based key registration works just fine, though. This is limited specifically to NFC.

**Comment by santiagorodriguez96** [comment]
Hey @neatchee! 

Is this is still an issue to you? 

I'm able to register using a YubiKey 5C with `CTAP2` enabled in Firefox in [webauthn.cedarcode.com](https://webauthn.cedarcode.com/session/new) (which is the [demo site for the gem `webauthn-ruby`](https://github.com/cedarcode/webauthn-rails-demo-app))

**Comment by neatchee** [comment]
@santiagorodriguez96 Still doesn't work on my instance (urusai.social) on Android Chrome. The demo site above doesn't even offer NFC for registering the key; it only offers USB.

**Comment by santiagorodriguez96** [comment]
@neatchee Ahhh I see, now I can reproduce it.

I'll take a look at what's going on 🤔
